packages/notifi-web-push-service-worker/lib/notifi-service-worker.js

```diff
@@ -0,0 +1,201 @@
+        store.delete(key);
+      });
+    },
+  };
+}
+
+self.addEventListener('install', function (event) {
+  self.skipWaiting();
+})
+
+function urlBase64ToUint8Array(base64String) {
+  var padding = '='.repeat((4 - base64String.length % 4) % 4);
+  var base64 = (base64String + padding).replace(/-/g, '+').replace(/_/g, '/');
+
+  var rawData = self.atob(base64);
+  var outputArray = new Uint8Array(rawData.length);
+
+  for (var i = 0; i < rawData.length; ++i) {
+    outputArray[i] = rawData.charCodeAt(i);
+  }
+  return outputArray;
+}
+
+async function createWebPushTarget(subscription, vapidPublicKey) {
+  try {
+    const subscriptionJson = subscription.toJSON();
+
+    const targetGroups = await client.getTargetGroups();
+    const defaultTargetGroup = targetGroups.find((targetGroup) => targetGroup.name === 'Default');
+    const webPushTargetIds = defaultTargetGroup?.webPushTargets?.map((t) => t?.Id) ?? [];
+
+    const webPushTargetResponse = await client.createWebPushTarget({
+      vapidPublicKey: vapidPublicKey,
+      endpoint: subscriptionJson.endpoint,
+      auth: subscriptionJson.keys.auth,
+      p256dh: subscriptionJson.keys.p256dh
+    })
+
+    if (!webPushTargetResponse.createWebPushTarget.webPushTarget) {
+      console.error('Failed to create web push target. CreateWebPushTargetMutation failed.');
+      return;
+    }
+
+    webPushTargetIds.push(webPushTargetResponse.createWebPushTarget.webPushTarget?.Id);
+
+    await client.ensureTargetGroup({
+      name: 'Default',
+      emailAddress: defaultTargetGroup?.emailTargets[0]?.emailAddress,
+      phoneNumber: defaultTargetGroup?.smsTargets[0]?.phoneNumber,
+      telegramId: defaultTargetGroup?.telegramTargets[0]?.telegramId,
+      discordId: defaultTargetGroup?.discordTargets[0]?.name,
+      slackId: defaultTargetGroup?.slackChannelTargets[0]?.name,
+      walletId: defaultTargetGroup?.web3Targets[0]?.name,
+      webPushTargetIds: webPushTargetIds
+    });
+  }
+  catch (err) {
+    console.error(err, "Failed to create web push target.")
+  }
+}
+
+function GetSubsciption(userAccount, dappId, env) {
+  if (Notification.permission !== "granted") {
+    console.log('Notification permissions not granted');
+    return;
+  }
+
+  if (!userAccount || !dappId || !env || userAccount == '' || dappId == '' || env == '') {
+    console.log('UserAccount, Notifi dappId, or env not found. Skipping subscription instantiation.');
+    return;
+  }
+
+  // TODO: Instantiate Notifi client here. If it fails, don't do anything.
+  client = instantiateFrontendClient(
+    dappId,
+    {
+      walletBlockchain: 'OFF_CHAIN',
+      userAccount: userAccount
+    },
+    env,
+    undefined,
+    { fetch }
+  );
+
+  client.initialize().then(userState => {
+    if (userState.status == 'authenticated') {
+      // TODO: Get vapid key here
+      let vapidPublicKey = "BBw1aI15zN4HFMIlbWoV2E390hxgY47-mBjN41Ewr2YCNGPdoR3-Q1vI-LAyfut8rqwSOWrcBA5sA5aC4gHcFjA";
```

Comment:
I need to add another endpoint here

Reply:
endpoint for backend API or endpoint in client side to obtain the public key to send to the service?

Reply:
Oh, I see. Nevermind

Reply:
yep